### DISCUSSION TITLE
New dataset regarding Influenza

### DISCUSSION THREAD

**Discussion by lisphilar** [lisphilar/covid19-sir#469] [Ideas]
## What dataset you need?
The number of cases with Influenza.
- Vaccinated
- Tested
- Confirmed (Tested positive)
- Closed (Recovered/Fatal)

## How will you use the dataset for your analysis?
- To update the models with Influenza dataset
- To predict the number of cases with influenza or COVID-19

This is from #244.

**Comment by Inglezos** [comment]
This proves to be a tricky issue. I would expect that such data would be readily available but for some reason they are not :/
I found this interesting paper:
https://www.nature.com/articles/s41598-020-80842-9

For their purposes they extracted influenza data manually from FluNet reports and they provide their code in R at the Supplementary Information section:
![image](https://user-images.githubusercontent.com/36845191/104322312-7dd49c00-54ed-11eb-8480-4bf50270a82c.png)

Could we implement something similar if we can't find any other way to find influenza datasets?

**Reply by lisphilar** [reply]
Thank you for your information.
I expected many databases will be created for influenza as well as COVID-19, but it is not actual. COVID-19 is outbreaking and influenza is hided.

I read the supplimentary file (Word file). The authors recommend scraping with `RSelenium` package. Data source is https://apps.who.int/flumart/Default?ReportNo=46

However, scraping is not a good solution for us because users may stress on the server with this solution nintentionally. (We have about 500 downloads/stable version in PyPI fortunately. https://pepy.tech/project/covsirphy)

One solution could be that we download the dataset from the server under CC licences and upload to GitHub repository. This seems under [CC BY-NC-SA 3.0 IGO](https://creativecommons.org/licenses/by-nc-sa/3.0/igo/), but we must consider the license of this database carefully. https://www.who.int/about/who-we-are/publishing-policies/copyright

**Reply by Inglezos** [reply]
>However, scraping is not a good solution for us because users may stress on the server with this solution nintentionally
>
First of all we need to determine the main reason why we want the influenza dataset. If we need to model influenza cases with SIR for example, in day-by-day basis then you are right we cannot do scraping.
But if we want just to model the cases per year or per half year or even per month **in order to compare the flu epidemic with coronavirus pandemic per country**, then it is okay to do that once per year or per half year or per month respectively (not the users but the maintainer of this package).

I thought we wanted mostly to implement the second case, since there is no API for influenza currently per country and that would be difficult to do so regularly (per day).

Do you want to extend CovsirPhy to modelling other illnesses as well, starting with influenza? Isn't this outside the scope/concept of this project?

If we want the first case and to extend this package to modelling other illnesses cases daily starting with influenza, yes we need to examine the license carefully and proceed. Perhaps we should reach the organization via email and ask them about both the license and for the dataset itself and if they are willing to provide us with the whole datasets in a more accessible way for our convenience.
